tfjs-backend-webgpu/src/kernels/maxpool_webgpu.ts

```diff
@@ -69,14 +74,36 @@ export class MaxPoolProgram implements WebGPUProgram {
 
             for (int wC = 0; wC < filterDims.x; wC += dilation.x) {
               int xC = xCCorner + wC * dilation.x;
-              float value = getValue(batch, xR, xC, d);
-              minMaxValue = max(value, minMaxValue);
+              for (int i = 0; i < ${this.workPerThread}; i++)
+              {
+                int d = coords[3] * ${this.workPerThread} + i;
+                if (d < outShape[3])
+                {
+                  float value = getValue(batch, xR, xC, d);
+                  minMaxValue[i] = max(value, minMaxValue[i]);
+                }
+                else
+                {
+                  break;
+                }
+              }
+            }
+          }
+          for (int i = 0; i < ${this.workPerThread}; i++)
+          {
+            int d = coords[3] * ${this.workPerThread} + i;
+            if (d < outShape[3])
+            {
+              setOutput(batch, coords[1], coords[2], d, minMaxValue[i]);
+            }
+            else
+            {
+              break;
             }
           }
-          setOutput(batch, coords[1], coords[2], d, minMaxValue);
         }
       }
     `;
-   this.shaderKey = 'maxpool';
+    this.shaderKey = 'maxpool';
```

Comment:
Wouldn't we need to add `this.workPerThread` to the `shaderKey`?

Reply:
Currently, `this.workPerThread` is fixed. So it won't affect the `shaderKey`. But in case  that we forget to change this place in future when we add dynamically to change `this.workPerThread`, it's ok to change it right now. Thanks.
@annxingyuan Please take another look. Thanks.